### PR TITLE
[Enhancement] (nereids)implement showCreateTableCommand in nereids

### LINKED ISSUE #42736: [Enhancement] (nereids)implement showCreateTableCommand in nereids
showCreateTable command is supported in legacy planner but not in nereids(new doris fe planner). you need implement the following showCreateTable command in nereids:
unsupportedshowCreateTableStatement
``` SHOW BRIEF? CREATE TABLE name=multipartIdentifier                             ```

### PR DESCRIPTION
**Pull request by msridhar78**
Issue Number: close #42736 

implement showCreateTableCommand in nereids

Changed  Session  Variables:
VarName                                                                                |  CurrentValue                |  DefaultValue
-----------------------------------------------|---------------------|-------------
character_set_results                                                    |  NULL                                |  utf8mb4          
sql_mode                                                                              |  STRICT_TRANS_TABLES  |                          
variable_version                                                              |  0                                      |  300                  
experimental_enable_nereids_distribute_planner  |  false                              |  true                

2024-11-18 11:17:06.052 INFO [suite-thread-1] (Suite.groovy:387) - Execute sql: DROP TABLE IF EXISTS `tb_show_create_table`
2024-11-18 11:17:06.070 INFO [suite-thread-1] (ScriptContext.groovy:120) - Run test_show_create_table_nereids in /root/doris/doris/regression-test/suites/nereids_p0/show/test_show_create_table_nereids.groovy succeed
2024-11-18 11:17:25.342 INFO [main] (RegressionTest.groovy:281) - Start to run single scripts
2024-11-18 11:18:43.673 INFO [main] (RegressionTest.groovy:410) - Success suites:
/root/doris/doris/regression-test/suites/nereids_p0/show/test_show_create_table_nereids.groovy: group=query,arrow_flight_sql,p0, name=test_show_create_table_nereids
2024-11-18 11:18:43.691 INFO [main] (RegressionTest.groovy:492) - All suites success.

### CODE REVIEW

**Comment by msridhar78** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateTableCommand.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    public String getTableName() {
+        return tableName;
+    }
+
+    public boolean isBrief() {
+        return isBrief;
+    }
+
+    @Override
+    public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateTableCommand.java:97]: ShowCreateTableStmt's analyze method needs to be re-implemented in nereids. For example, add a `private void validate(ConnectContext ctx) throws UserException` method to do the job, and call it at the beginning of run method.

Comment (msridhar78) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateTableCommand.java:135]: Done - please review

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateTableCommand.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    @Override
+    public <R, C> R accept(PlanVisitor<R, C> visitor, C context) {
+        return visitor.visitShowCreateTableCommand(this, context);
+    }
+
+    @Override
+    public ShowResultSet doRun(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        validate(ctx);
+        // If dbName is not provided, use the default database from the context
+        String effectiveDbName = (tblNameInfo.getDb() == null || tblNameInfo.getDb().isEmpty())
```
Comment (starocean999) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateTableCommand.java:121]: please refer to handleShowCreateTable() in ShowExecutor.java. The old code just needs some minor changes, it's easier to reivew. thanks.

Comment (msridhar78) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateTableCommand.java:117]: done - please review

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateTableCommand.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+                            ? ctx.getDatabase() : tblNameInfo.getDb();
+
+        // If catalog is not provided, use the default catalog (e.g., internal)
+        String effectiveCatalog = (tblNameInfo.getCtl() == null || tblNameInfo.getCtl().isEmpty())
```
[the remark this answers is not in the document]
Comment (msridhar78) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateTableCommand.java]: done

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateTableCommand.java [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    @Override
+    public <R, C> R accept(PlanVisitor<R, C> visitor, C context) {
+        return visitor.visitShowCreateTableCommand(this, context);
+    }
+
+    @Override
+    public ShowResultSet doRun(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        validate(ctx);
+        // If dbName is not provided, use the default database from the context
+        String effectiveDbName = (tblNameInfo.getDb() == null || tblNameInfo.getDb().isEmpty())
```
[the remark this answers is not in the document]
Comment (msridhar78) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateTableCommand.java]: done - pls review

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateTableCommand.java [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+                            ? ctx.getDatabase() : tblNameInfo.getDb();
+
+        // If catalog is not provided, use the default catalog (e.g., internal)
+        String effectiveCatalog = (tblNameInfo.getCtl() == null || tblNameInfo.getCtl().isEmpty())
+                            ? "internal" : tblNameInfo.getCtl();
+
+        // Fetch the catalog, database, and table metadata
+        Env env = Env.getCurrentEnv();
+        DatabaseIf db = env.getCatalogMgr().getCatalogOrAnalysisException(effectiveCatalog)
```
[the remark this answers is not in the document]
Comment (msridhar78) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/ShowCreateTableCommand.java]: done - pls review

**Comment by msridhar78** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
pls resolve conflicts and regression cases fail

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run buildall

**Comment by msridhar78** [comment]
run performance

**Comment by msridhar78** [comment]
run p0

**Comment by msridhar78** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Pull request merged**